calculators/gas_king_rating

```diff
@@ -0,0 +1,101 @@
+<?php
+
+$GLOBALS['rating_calculators']['gas_king_rating'] = [
+    'description' => '
+        gas king rating
+        only objective: gas as much and as soon as you can
+
+        you start with 2000 rating (compression is activated although maybe not efficient, I was too lazy to see how you did it)
+        you lose 10 rating per minute for each gas not taken
+        you win 1, 2, 3, 4 rating per minute for gas 1, 2, 3, 4
+        ',
+    'default_rating' => 2000,
+    'fields' => [
+        'g.duration',
+        'gp.gas_timings',
+        'gp.gas_count',
+        ],
+    'main' => 'dsr_gas_king_rating__main',
+    ];
+
+function same_sign($a, $b)
+{
+    return ($a >= 0 && $b >= 0) || ($a < 0 && $b < 0);
+}
+
+function dsr_gas_king_rating__main(&$teams)
+{
+    foreach($teams as &$team)
+    {
+        foreach($team as &$player)
+        {
+            $diff = 0;
+            $duration_of_the_game_in_minutes = $teams[0][0]['duration'] / 22.4 / 60;
+
+            if($player['gas_count'] === 0)
```

Comment:
does it work? I think field values such as this come in a string format, therefore you'll have to use == 0 comparison here

Reply:
I honestly have no idea since this is my first code in php.

Reply:
I tried with intval and without it and it didn't change any rating from what I've seen

Reply:
gas_count is actually a variable that is used for faster database queries. In code there's no real need to use it, you could use count($player['gas_timings']), but using it is fine too.